gnovm/pkg/gnolang/op_expressions.go

```diff
@@ -194,8 +194,13 @@ func (m *Machine) doOpRef() {
 			nv.Value = rv2
 		}
 	}
+	// when obtaining a pointer of the databyte type, use the ElemType of databyte
+	elt := xv.TV.T
+	if elt == DataByteType {
+		elt = xv.TV.V.(DataByteValue).ElemType
```

Comment:
Nice work!  Everything works as expected. 

This line of code,  solve another issue that is not directly related to the comparison issues in this PR 

```
	a := []byte("hello")
	a1 := &a[1]
	b1 := *a1 
        println(b1)

interface conversion: gnolang.Value is nil, not gnolang.DataByteValue
```
there are some other things we need to take care of to solve the issue completely. We will write a separate PR

The rest LGTM